.github/workflows/build-images-ci.yaml

```diff
@@ -22,7 +22,20 @@ on:
     types:
      - completed
 
-permissions: read-all
+permissions:
+  actions: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: write
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  pull-requests: read
+  repository-projects: read
+  security-events: read
+  statuses: read
```

Comment:
Can you elaborate on why is this now necessary? 🤔

Reply:
Ah I see, `cosign` uses OIDC. Thanks!

As for previous permissions, we need to do a test run but I believe we only need `contents` here (for `actions/checkout`).

Reply:
Should I update the PR with only the `contents: read` and `id-token: write` permissions or should I wait until we find a way to do a test run?

Reply:
These permissions should be the minimal necessary to implement the functionality if we only need `contents` then let's try that.

The test run is being taken care of in https://github.com/cilium/cilium/pull/21435 right? We can sync that on-demand.

Reply:
Thank you. Updated PR with the suggested changes.

Reply:
Test run seems to be successful with the current updated permissions.